tools/build_langserver/langserver/handler.go

```diff
@@ -137,25 +141,24 @@ func (h *LsHandler) handleExit(ctx context.Context, request *jsonrpc2.Request) (
 }
 
 func (h *LsHandler) handleCancel(ctx context.Context, request *jsonrpc2.Request) (result interface{}, err error) {
-	// TODO(bnmetrics): rethink this, Try and do something with the request id
-	/*
-		**comments from Pebers**:
-				Yeah it looks like it will need more complexity -
-				like the server will need to keep a map of request id -> context cancel function,
-				and this function would cancel the appropriate one.
-
-				Tbh sounds like a classic thing we can put off for later :)
-	 */
-	if request.Params == nil {
+	// Is there is no param with Id, or if there is no requests stored currently, return nothing
+	if request.Params == nil || h.requestStore.requests == nil {
```

Comment:
why would `h.requestStore.requests` be nil? doesn't seem like something you should have to worry about here.

Reply:
Oh! I mostly wanted to check if the map is empty

Reply:
nil isn't the same as an empty map :)

But I think that map should be totally encapsulated by the requestStore object - it would be nicer if the handler only interacted with it through its public interface.